service/history/queue/cross_cluster_queue_processor_base.go

```diff
@@ -306,14 +306,23 @@ func (c *crossClusterQueueProcessorBase) completeAckTasks(nextAckLevel int64) (i
 	ctx, cancel := context.WithTimeout(context.Background(), taskCleanupTimeout)
 	defer cancel()
 
-	if err := c.shard.GetExecutionManager().RangeCompleteCrossClusterTask(ctx, &persistence.RangeCompleteCrossClusterTaskRequest{
-		TargetCluster:        c.targetCluster,
-		ExclusiveBeginTaskID: c.ackLevel,
-		InclusiveEndTaskID:   nextAckLevel,
-	}); err != nil {
-		return c.ackLevel, err
+	for {
+		pageSize := c.options.DeleteBatchSize()
+		resp, err := c.shard.GetExecutionManager().RangeCompleteCrossClusterTask(ctx, &persistence.RangeCompleteCrossClusterTaskRequest{
+			TargetCluster:        c.targetCluster,
+			ExclusiveBeginTaskID: c.ackLevel,
+			InclusiveEndTaskID:   nextAckLevel,
+			PageSize:             pageSize, // pageSize may or may not be honored
+		})
+		if err != nil {
+			return c.ackLevel, err
+		}
+		if resp.TasksCompleted < pageSize || // all target tasks are deleted
+			resp.TasksCompleted == persistence.UnknownNumRowsAffected || // underlying database does not support rows affected, so pageSize is not honored and all target tasks are deleted
+			resp.TasksCompleted > pageSize { // pageSize is not honored and all tasks are deleted
+			break
```

Comment:
nit: can we create some util method for this logic, since it's a bit complicated.  Duplicating it for each queue processor is kind of error-prone when we need to modify it. 

I guess if the only why we use the returned response is through this check, we can also just return if the deletion is completed?